pkg/controller/association/reconciler.go

```diff
@@ -388,7 +388,10 @@ func (r *Reconciler) updateStatus(ctx context.Context, associated commonv1.Assoc
 	newStatus = associated.AssociationStatusMap(r.AssociationType)
 
 	if !reflect.DeepEqual(oldStatus, newStatus) {
-		if err := r.Status().Update(associated); err != nil {
+		if err := associated.SetAssociationStatusMap(r.AssociationType, newStatus); err != nil {
```

Comment:
Why are we setting the status once more here?

Reply:
I think that's a merge gone awry. Good spot!